apis/python/src/tiledbsoma/_soma_object.py

```diff
@@ -335,3 +344,37 @@ def exists(
 
 
 AnySOMAObject = SOMAObject[_tdb_handles.AnyWrapper]
+
+
+def _read_soma_type(hdl: _tdb_handles.AnyWrapper) -> str:
+    obj_type = hdl.metadata.get(SOMA_OBJECT_TYPE_METADATA_KEY)
+    encoding_version = hdl.metadata.get(SOMA_ENCODING_VERSION_METADATA_KEY)
+
+    if obj_type is None:
+        raise SOMAError(
+            f"stored TileDB object does not have {SOMA_OBJECT_TYPE_METADATA_KEY!r}"
+        )
+
+    if isinstance(obj_type, bytes):
+        obj_type = str(obj_type, "utf-8")
+
+    if not isinstance(obj_type, str):
+        raise SOMAError(
+            f"stored TileDB object {SOMA_OBJECT_TYPE_METADATA_KEY!r}"
+            f" is {type(obj_type)}"
+        )
+    if encoding_version is None:
+        raise SOMAError(
+            f"stored TileDB object does not have {SOMA_ENCODING_VERSION_METADATA_KEY}"
+        )
+
+    if isinstance(encoding_version, bytes):
+        encoding_version = str(encoding_version, "utf-8")
+
+    if encoding_version not in SUPPORTED_SOMA_ENCODING_VERSIONS:
+        raise ValueError(
+            f"Unsupported SOMA object encoding version {encoding_version}. The TileDB-SOMA "
+            f"client library needs to be updated to a more recent version."
```

Comment:
```suggestion
            f"Unsupported SOMA object encoding version {encoding_version}. TileDB-SOMA "
            f"needs to be updated to a more recent version."
```
This one was mine. In retrospect I think specifying "client library" is more likely to be confusing than helpful.